### ISSUE TITLE
error: unknown type name

### ISSUE THREAD

**Issue by akira-okumura** [ROBAST/ROBAST#28]
The simple test script below emits `error: unknown type name` when it is loaded twice. ROOT 6.14.06 and ROBAST 2.5.0. If the include header line is uncommented, it works as expected. But including header files is not necessary for the standard ROOT classes.

```
//#include "AOpticsManager.h"

void test() {
  AOpticsManager* manager = new AOpticsManager("manager", "manager");
}
```

```
$ root
root [0] .x test.C
 ----------------------------------------------------------------------------
| Welcome to ROBAST                               https://robast.github.io/  |
|                                                                            |
| Please cite the following paper when you publish your ROBAST simulation.   |
|                                                                            |
| Akira Okumura, Koji Noda, Cameron Rulten (2016)                            |
| "ROBAST: Development of a ROOT-Based Ray-Tracing Library for Cosmic-Ray    |
| Telescopes and its Applications in the Cherenkov Telescope Array"          |
| Astroparticle Physics 76 38-47                                             |
|                                                                            |
| For support & FAQ, please visit https://robast.github.io/support.html      |
|                                                                            |
|                       ROBAST is developed by Akira Okumura (oxon@mac.com)  |
 ----------------------------------------------------------------------------
Info in <TGeoManager::TGeoManager>: Geometry manager, manager created
root [1] .x test.C
In file included from input_line_27:1:
/Users/oxon/test.C:4:3: error: unknown type name 'AOpticsManager'
  AOpticsManager* manager = new AOpticsManager("manager", "manager");
  ^
/Users/oxon/test.C:4:33: error: unknown type name 'AOpticsManager'
  AOpticsManager* manager = new AOpticsManager("manager", "manager");
                                ^
```

I remember this did not happen a while ago (ROOT 5?).

**Comment by akira-okumura** [comment]
It also appears on CentOS 7 with ROOT 6.14/06 but not on El Capitan with ROOT 5.34/38. It looks to be a ROOT 6 problem.

**Comment by micheledoro** [comment]
We are also trying to test the installation wtih High Sierra here in Padova, I will let you know if it works.

**Comment by akira-okumura** [comment]
Thanks. I have post this issue on the ROOT forum too.
https://root-forum.cern.ch/t/error-unknown-type-name-in/31980

**Comment by akira-okumura** [comment]
According to Axel the issue still exits in the ROOT 6 repo and they are working on it.
https://root-forum.cern.ch/t/error-unknown-type-name-in/31980/3?u=oxon

We need a ROBAST side solution to avoid it if possible. But
```
root [0] .x test.C
root [1] #include "AOpticsManager.h"
root [2] .x test.C
```
didn't help.

**Comment by akira-okumura** [comment]
Adding `gInterpreter->Declare("#include \"AOpticsManager.h\"");` in a `__attribute__((constructor)) static void` function didn't help. It looks that `include` must be done before loading the ROBAST library.

**Comment by akira-okumura** [comment]
Still exists in 6.16/00

**Comment by akira-okumura** [comment]
It looks the issue has been fixed in 6.18/04
